### DISCUSSION TITLE
V1 CAG -> V2 geom2

### DISCUSSION THREAD

**Discussion by napter** [jscad/OpenJSCAD.org#956] [Q&A]
Hi, I am trying to upgrade from V1 to V2. In the previous code I call: CAG.frompoints with an array of paths (array of arrays of points). The CAG from points would handle this and when an array of array was sent it would call fromNestedPointsArray ([see here](https://github.com/jscad/csg.js/blob/054604309613a40b500f040169b8cd8ed9413343/src/core/CAGFactories.js#L40)). 

In the V2 code it seems the equivalent code would be geom2.frompoints. However, this function only takes an array of points. Is there an other way to make the 2d geometry from an array of points like there was in V1?

**Comment by z3dev** [comment]
@napter good question!

actually, that logic was for duplicating OPENSCAD functionality. So, the logic did not become part of V2 core functionality.

you could very easily lift the V1 logic to make a small function.

**Comment by z3dev** [comment]
There’s a related issue #26 which was about supporting the SCAD functionality.

I was thinking that @platypii could add this to the conversion script.
